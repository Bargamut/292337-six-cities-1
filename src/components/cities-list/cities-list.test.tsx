import * as React from 'react';
import renderer from 'react-test-renderer';
import CitiesList from './cities-list.jsx';

const mock = {
  cities: [
    `Saint-Petersburg`,
    `Amsterdam`,
    `Berlin`,
    `London`,
    `Budapest`,
    `Tokyo`,
    `Sydney`
  ]
};

it(`Cities List correctly renders`, () => {
  const {cities} = mock;

  const citiesList = renderer.create(
      <CitiesList
        city="Amsterdam"
        cities={cities}
        onChangeCity={jest.fn()}
      />
  )
  .toJSON();

  expect(citiesList).toMatchSnapshot();
});
