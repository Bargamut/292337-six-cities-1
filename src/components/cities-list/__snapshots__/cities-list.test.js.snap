// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities List correctly renders 1`] = `
<div
  className="cities tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Saint-Petersburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          href="#"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Berlin
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            London
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Budapest
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Tokyo
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Sydney
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
